### PR TITLE
Polish MeterTag changes

### PR DESCRIPTION
**Pull request by izeye**
This PR polishes the `MeterTag` changes changes a bit.

See gh-3727

### CODE REVIEW

File: micrometer-commons/src/main/java/io/micrometer/common/annotation/AnnotationUtils.java [hunk 1]
```diff
@@ -34,14 +34,14 @@ private AnnotationUtils() {
...
-    static List<AnnotatedParameter> findAnnotatedParameters(Class<? extends Annotation> tagClazz, Method method,
+    static List<AnnotatedParameter> findAnnotatedParameters(Class<? extends Annotation> annotationClazz, Method method,
             Object[] args) {
         Annotation[][] parameters = method.getParameterAnnotations();
         List<AnnotatedParameter> result = new ArrayList<>();
         int i = 0;
         for (Annotation[] parameter : parameters) {
             for (Annotation parameter2 : parameter) {
-                if (tagClazz.isAssignableFrom(parameter2.getClass())) {
+                if (annotationClazz.isAssignableFrom(parameter2.getClass())) {
```
[the remark this answers is not in the document]
Comment (shakuzen) [reply micrometer-commons/src/main/java/io/micrometer/common/annotation/AnnotationUtils.java:44]: @marcingrzejszczak should we ignore this?

[the remark this answers is not in the document]
Comment (marcingrzejszczak) [reply micrometer-commons/src/main/java/io/micrometer/common/annotation/AnnotationUtils.java:44]: That's a good question. It's been like this for years in Sleuth (of course it doesn't have to mean anything)

[the remark this answers is not in the document]
Comment (jonatan-ivanov) [reply micrometer-commons/src/main/java/io/micrometer/common/annotation/AnnotationUtils.java:44]: I think it makes sense using `annotationType()` here.

**Comment by marcingrzejszczak** [comment]
@izeye can you check if following this commit (https://github.com/micrometer-metrics/micrometer-docs/commit/800a561360e9be564f9857572e88e5851a40600f ) anything there needs to be changed?

**Comment by izeye** [comment]
@marcingrzejszczak Sure, I created https://github.com/micrometer-metrics/micrometer-docs/pull/300.

**Pull request merged**